core/src/main/java/org/elasticsearch/index/analysis/WordDelimiterGraphTokenFilterFactory.java

```diff
@@ -0,0 +1,101 @@
+/*
```

Comment:
Ok I created https://github.com/elastic/elasticsearch/issues/23104 to track the inclusion of this new filter. We can add the documentation and tests in a follow-up